x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -314,7 +330,35 @@ public NamedExpression visitQualifiedNamePattern(EsqlBaseParser.QualifiedNamePat
                 objects.add(".");
             }
 
-            String patternContext = patterns.get(i).ID_PATTERN().getText();
+            String patternContext = "";
+            if (patterns.get(i).ID_PATTERN() != null) {
+                patternContext = patterns.get(i).ID_PATTERN().getText();
+            } else if (patterns.get(i).params() != null) {
+                Expression exp = expression(patterns.get(i).params());
+                if (exp instanceof Literal lit) {
+                    // only Literal.NULL can happen with missing params, params for constants are not allowed
+                    if (lit.value() != null) {
+                        throw new ParsingException(src, "Constant [{}] is unsupported for [{}]", patterns.get(i), ctx.getText());
+                    }
+                } else if (exp instanceof UnresolvedAttribute ua) {
+                    String identifier = ua.name();
+                    if (identifier != null) {
+                        if (identifier.contains(WILDCARD) && (identifier.charAt(0) == '`') == false) {
+                            // identifier with star need to be quoted
+                            patternContext = "`" + identifier.replace("`", "``") + "`";
+                        } else {
+                            patternContext = identifier;
```

Comment:
This is incorrect - what if the identifier is a followed by backtick or simply backtick?

Reply:
We can probably just skip the code to process fragment, if this is an identifier(`UnresolvedAttribute`), to avoid this headache?